### PR TITLE
Add class to cast types

### PR DESCRIPTION
**Pull request by ghost**
Add `CAST` function from https://gist.github.com/galiazzi/5e5f04f9753ba4d8a9b972c87dc2a805

### CODE REVIEW

File: src/Cast.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public function getSql(SqlWalker $sqlWalker)
+    {
+        $type = trim($this->expr2->dispatch($sqlWalker), "'");
+        return sprintf(
+            'cast(%s as %s)',
+            $this->expr1->dispatch($sqlWalker),
+            $type
+        );
+    }
+}
```
Comment (ajgarlag) [inline src/Cast.php]: Add a blank line at the end.

File: src/Cast.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+use Doctrine\ORM\Query\AST\Functions\FunctionNode;
+use Doctrine\ORM\Query\Lexer;
+use Doctrine\ORM\Query\Parser;
+use Doctrine\ORM\Query\SqlWalker;
+
+/**
+ * "CAST" "(" ArithmeticExpression "," StringPrimary ")".
+ *
+ * @copyright 2018-2020 Dayan C. Galiazzi
```
Comment (ajgarlag) [inline src/Cast.php:22]: Add a `@see` tag pointing to the origin URL

**Review by ajgarlag — CHANGES_REQUESTED** [review]

**Comment by ajgarlag** [comment]
@galiazzi I'd like to add your [`cast`](https://gist.github.com/galiazzi/5e5f04f9753ba4d8a9b972c87dc2a805) function to this repository, which is MIT licensed.
As there is no license information in your gist, do you approve the PR?
Thanks in advance.

**Comment by ajgarlag** [comment]
Closed due to lack of response of original code author.

See #2  instead

**Comment by galiazzi** [comment]
@ajgarlag  

Yes, I approve

I'm sorry for answering so late

**Comment by ajgarlag** [comment]
@galiazzi Finally I've written the CAST function inspired by `oro/doctrine-extensions` implementation.

Comparing it with your implementation, the resulting DQL is more similar to SQL standard: `CAST(field, 'text')` vs `CAST(field AS text)`.

Thanks anyway.

**Pull request closed without merging**